payments-ui-core/src/main/java/com/stripe/android/ui/core/elements/CardNumberController.kt

```diff
@@ -31,12 +56,29 @@ internal class CardNumberController constructor(
     }
 
     private val _fieldState = combine(cardBrandFlow, _fieldValue) { brand, fieldValue ->
-        cardTextFieldConfig.determineState(brand, fieldValue)
+        cardTextFieldConfig.determineState(
+            brand,
+            fieldValue,
+            accountRangeService.accountRange?.panLength ?: brand.getMaxLengthForCardNumber(fieldValue)
+        )
     }
     override val fieldState: Flow<TextFieldState> = _fieldState
 
     private val _hasFocus = MutableStateFlow(false)
 
+    @VisibleForTesting
+    val accountRangeService = CardAccountRangeService(
```

Comment:
This is really cool, now all the logic for the service is tied up nicely in a bow in this CardAccountRangeService